tests/raftstore/test_snap.rs

```diff
@@ -288,7 +288,7 @@ struct StaleSnap {
 }
 
 impl Filter<RaftMessage> for Arc<StaleSnap> {
-    fn before(&self, msgs: &mut Vec<RaftMessage>) {
+    fn before(&self, msgs: &mut Vec<RaftMessage>) -> FilterResult {
```

Comment:
1. Result is also an enum.
2. Now you return the result for every Filter::before, so the filter can check whether return ok or error itself, if error, `send` returns error, we don't need to check the message is empty and whether we should return DroppableError or not again. 
3. If the filter meets an error but want to go on, it can return ok. 
4. We can check error if we really want to filter some cases later, but I think we don't need this.